### PR TITLE
Update encryption-at-rest.md

### PR DESCRIPTION
**Pull request by jiayang-zheng**
Add SM4 support in v6.3.0.

### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)

Add SM4 support in v6.3.0.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [x] v6.3 (TiDB 6.3 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/10563, https://github.com/pingcap/docs-cn/pull/11277, [encryption-at-rest](https://docs.pingcap.com/zh/tidb/dev/encryption-at-rest).    
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: encryption-at-rest.md [hunk 1]
```diff
@@ -91,7 +93,7 @@ data-encryption-method = "aes128-ctr"
...
+Possible values for `data-encryption-method` are "aes128-ctr", "aes192-ctr", "aes256-ctr", "sm4-ctr" (only in v6.3.0 and later versions) and "plaintext". The default value is "plaintext", which means encryption is not turned on. `data-key-rotation-period` defines how often TiKV rotates the data key. Encryption can be turned on for a fresh TiKV cluster, or an existing TiKV cluster, though only data written after encryption is enabled is guaranteed to be encrypted. To disable encryption, remove `data-encryption-method` in the configuration file, or reset it to "plaintext", and restart TiKV. To change encryption method, update `data-encryption-method` in the configuration file and restart TiKV.
```
Comment (shichun-0415) [inline encryption-at-rest.md:96]: The following info in the [Chinese PR](https://github.com/pingcap/docs-cn/pull/10563/files#diff-aca7ce975f5b852a634dc43e7d7b22d1af3a250516efbaeaa5f1ee24cc5bb680R94) is missing in the English version:
> 替换加密算法后，旧加密算法生成的加密文件会随着新数据的写入逐渐被重写成新加密算法所生成的加密文件。

File: encryption-at-rest.md [hunk 2]
```diff
@@ -20,14 +20,16 @@ When a TiDB cluster is deployed, the majority of user data is stored on TiKV and
...
 
 Optionally, you can use AWS KMS for both cloud and on-premises deployments. You can also supply the plaintext master key in a file.
 
 TiKV currently does not exclude encryption keys and user data from core dumps. It is advised to disable core dumps for the TiKV process when using encryption at rest. This is not currently handled by TiKV itself.
 
 TiKV tracks encrypted data files using the absolute path of the files. As a result, once encryption is turned on for a TiKV node, the user should not change data file paths configuration such as `storage.data-dir`, `raftstore.raftdb-path`, `rocksdb.wal-dir` and `raftdb.wal-dir`.
 
+SM4 encryption is only supported on v6.3.0 and later versions of TiKV, and TiKV versions before v6.3.0 only support AES encryption.
```
Comment (TomShawn) [inline encryption-at-rest.md:31]:
```suggestion
SM4 encryption is only supported in v6.3.0 and later versions of TiKV. TiKV versions before v6.3.0 only support AES encryption.
```

File: encryption-at-rest.md [hunk 3]
```diff
@@ -50,7 +52,7 @@ TiKV, TiDB, and PD  info logs might contain user data for debugging purposes. Th
 
 ### Overview
 
-TiKV currently supports encrypting data using AES128, AES192 or AES256, in CTR mode. TiKV uses envelope encryption. As a result, two types of keys are used in TiKV when encryption is enabled.
+TiKV currently supports encrypting data using AES128, AES192, AES256 or SM4 (only in v6.3.0 and later versions), in CTR mode. TiKV uses envelope encryption. As a result, two types of keys are used in TiKV when encryption is enabled.
```
Comment (TomShawn) [inline encryption-at-rest.md:55]:
```suggestion
TiKV currently supports encrypting data using AES128, AES192, AES256, or SM4 (only in v6.3.0 and later versions), in CTR mode. TiKV uses envelope encryption. As a result, two types of keys are used in TiKV when encryption is enabled.
```

File: encryption-at-rest.md (hunk 1 above)
Comment (TomShawn) [reply encryption-at-rest.md:96]:
```suggestion
Possible values for `data-encryption-method` are "aes128-ctr", "aes192-ctr", "aes256-ctr", "sm4-ctr" (only in v6.3.0 and later versions) and "plaintext". The default value is "plaintext", which means encryption is not turned on. `data-key-rotation-period` defines how often TiKV rotates the data key. Encryption can be turned on for a fresh TiKV cluster, or an existing TiKV cluster, though only data written after encryption is enabled is guaranteed to be encrypted. To disable encryption, remove `data-encryption-method` in the configuration file, or reset it to "plaintext", and restart TiKV. To change encryption method, update `data-encryption-method` in the configuration file and restart TiKV. To change the encryption algorithm, replace `data-encryption-method` with a supported encryption algorithm and then restart TiKV. After the replacement, as new data is written, the encryption file generated by the previous encryption algorithm is gradually rewritten to a file generated by the new encryption algorithm.
```

**Review by TomShawn — APPROVED** [review]
LGTM

File: encryption-at-rest.md [hunk 4]
```diff
@@ -20,14 +20,16 @@ When a TiDB cluster is deployed, the majority of user data is stored on TiKV and
...
 
 Optionally, you can use AWS KMS for both cloud and on-premises deployments. You can also supply the plaintext master key in a file.
 
 TiKV currently does not exclude encryption keys and user data from core dumps. It is advised to disable core dumps for the TiKV process when using encryption at rest. This is not currently handled by TiKV itself.
 
 TiKV tracks encrypted data files using the absolute path of the files. As a result, once encryption is turned on for a TiKV node, the user should not change data file paths configuration such as `storage.data-dir`, `raftstore.raftdb-path`, `rocksdb.wal-dir` and `raftdb.wal-dir`.
 
+SM4 encryption is only supported in v6.3.0 and later versions of TiKV. TiKV versions earlier than v6.3.0 only support AES encryption.
```
Comment (shichun-0415) [inline encryption-at-rest.md:31]:
```suggestion
SM4 encryption is only supported in v6.3.0 and later versions of TiKV. TiKV versions earlier than v6.3.0 only support AES encryption. SM4 encryption may lead to 50% to 80% degradation on throughput performance. 
```

File: encryption-at-rest.md [hunk 5]
```diff
@@ -20,14 +20,16 @@ When a TiDB cluster is deployed, the majority of user data is stored on TiKV and
...
+TiKV supports encryption at rest. This feature allows TiKV to transparently encrypt data files using [AES](https://en.wikipedia.org/wiki/Advanced_Encryption_Standard) or [SM4](https://en.wikipedia.org/wiki/SM4_(cipher)) in [CTR](https://en.wikipedia.org/wiki/Block_cipher_mode_of_operation) mode. To enable encryption at rest, an encryption key must be provided by the user and this key is called master key. TiKV automatically rotates data keys that it used to encrypt actual data files. Manually rotating the master key can be done occasionally. Note that encryption at rest only encrypts data at rest (namely, on disk) and not while data is transferred over network. It is advised to use TLS together with encryption at rest.
```
[the remark this answers is not in the document]
Comment (TomShawn) [reply encryption-at-rest.md:23]: done

**Pull request merged**